proposals/selfAssemblingModelHierarchy/README.md

```diff
@@ -0,0 +1,342 @@
+
+> When assets are published/packaged with model kinds already established, the model hierarchy becomes mostly “self assembling”.
+
+As the above examples demonstrate, a self assembling model hierarchy unfortunately has been complicated by the need for the "glue" `group` kind.
+
+There is prior art for composition "fixing" errant model hierarchy. Consider just the department store. It's not uncommon to see users (incorrectly) apply the `group` kind to all `Xform` and `Scope` prims in a hierarchy.
+```
+def Xform "DepartmentStore" (kind = "component") {
+    def Scope "GroundFloor" (kind = "group") {
+        def Scope "Housewares" (kind = "subcomponent") {
+        }
+    }
+}
+```
+
+The authored `group` kind is ignored with respect to model hierarchy because its parent isn't a `group` or `assembly`. With minimal trade-offs, this proposal argues that just as incorrect usage can be discarded by composition, _correct `group` usage can be propagated by composition_.
+
+More simply-- **untagged children of groups and assemblies are automatically groups**.
+
+> **Note** This proposal frames `group` as being an "unimportant" bridge kind, but that tooling and pipelines may not have reasoned about `group` in the same way. Please read `group` propagation not as literally propagating the `kind` field onto descendants but just that the prim may contain descendants in the _model hierarchy_.
+
+When applied recursively, this simple change would correct most incorrect model hierarchy usage and vastly simplify maintenance. Model hierarchy would be "self assembling".
+
+Let's revise the example, removing sites where explicit authoring of `kind=group` would no longer be necessary.
+```
+def Scope "TriStateArea" (kind = "group") {
+    def Xform "NewYorkCity" (kind = "assembly") {
+        def Scope "Water" {}
+        def Scope "Bridges" {
+            def Xform "BrooklynBridge" (kind = "component") {}
+        }
+        def Scope "Tunnels" {
+            def Xform "LincolnTunnel" (kind = "component") {}
+        }
+        def Scope "Boroughs" {
+            def Scope "Manhattan" (kind = "assembly") {
+                def Scope "FifthAvenue" {
+                    def Xform "DepartmentStore" (kind = "component") {
+                        def Scope "GroundFloor" {
+                            def Scope "Housewears" (kind = "subcomponent") {
+                                def Xform "Aisle_1" {
+                                    def Mesh "TopShelf" {
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+In this example, because `NewYorkCity` is an `assembly` (and therefore a `group`), its descendant `Water`, `Tunnels`, `Bridges`, and `Boroughs` scopes are automatically `group` prims. Propagation would end at the `BrooklynBridge`, `LincolnTunnel`, and `DepartmentStore` `component` models.
+
```

Comment:
Before getting to discussion about performance, it would be very helpful to first give a modified example of above where there *is* some unpruned component-less path(s), and discuss whether, in terms of `IsGroup()`/`IsModel()` queries, what the prims along that path would be considered to be.  

Of particular concern to me is: as we are doing our current single population pass through the prims on the stage, is it still the case that we only compute a prim's "effective model kind" (to be cached in primFlags) **once** , or is there a need/desire to revisit ancestor prims after we reach a "model hierarchically unpruned leaf prim"?